tests/test_decision_gates.py

```
@@ -158,3 +160,23 @@ def test_katex_is_loaded_and_the_distribution_heading_is_math():
+
+def test_the_markout_row_reads_the_top_level_payload_fields():
+    # Arrange — `markout_samples` and `adverse_selection` live at the top level
+    # of the KPI payload, not inside trade_analytics. Reading only `ta` made
+    # this row report "0 samples" on runs that had measured plenty.
+    # trade_analytics deliberately carries the WRONG values, so the test fails
+    # if the renderer goes back to reading them from there.
+    ta = {**_healthy(), "markout_samples": 0, "adverse_selection": None}
+    payload = {"trade_analytics": ta, "statistical_analytics": {}, "n": 40,
+               "kpi": {"markout_samples": 30, "adverse_selection": 0.0022,
+                       "adverse_selection_excess": 0.0009}}
+    out = subprocess.run([shutil.which("node"), str(HARNESS), json.dumps(payload)],
```

Ruff (0.16.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)